File: new/dart-boxes/export-rows.dart
// 📦 קופסת-חיבורים · שורות-הייצוא (Dart) — מחווטת 9 אטומי-Dart. מקבילה ל-new/boxes/export-rows.mjs.
// חוזה משותף: new/boxes/export-rows.contract.md. מקור-האמת: maor/src/lib/exportRows.ts.
// זהו ההוכחה ש-מאור(JS) ובנייה-חכמה(Dart) מתחברות לאותה קופסה: אותם קלטים ⇒ אותו פלט.
//
// החיווט הגלוי (כבמקור-ה-JS): משפחות/תומכות = חוטים ישירים (בלי שקעים);
// אירועים = חוט + שרשרת-התאריך-העברי (hebDateFull על gem/gemYear/hebParts,
// סדר-שקעים כבמקור hebrew.ts:160) + מילון-הסוגים (EV_META) + termOf.
//
// ⚠️ גבול-פלטפורמה (חוק-6): toCsv/downloadCsv (DOM) אינם בקופסה — שקע של
//   שער-הייצוא/לוח-האם, מוזרק פר-פלטפורמה. מה שמוכח כאן = בניית-השורות, זהה-ביט.
import '../dart-maor/families-import-format-rows.dart' as fif;
import '../dart-maor/supporters-import-format-rows.dart' as sif;
import '../dart-maor/events-csv-rows.dart' as ecr;
import '../dart-maor/term-of.dart' as tof;
import '../dart-maor/heb-date-full.dart' as hdf;
import '../dart-maor/gematria.dart' as gm;
import '../dart-maor/gem-year.dart' as gy;
import '../dart-maor/heb-parts.dart' as hp;
import '../dart-maor/ev-meta.dart' as em;

// ── החיווט: שרשרת-התאריך-העברי — הכרעת-הקופסה (hebrew.ts:160) ──
// JS: const hebFull = (iso) => hebDateFull(iso, gem, (y) => gemYear(y, gem), hebParts);
// סדר-השקעים (iso, gem, gemYear(·,gem), hebParts) verbatim; gemYear מקבל את gem כשקע (חוק-1).
String _hebFull(dynamic iso) =>
    hdf.hebDateFull(iso as String?, gm.gem, (y) => gy.gemYear(y, gm.gem), hp.hebParts);

// ── החשיפה ──────────────────────────────────────────────────────────────────
/// משפחות בפורמט-הייבוא (13 עמודות) — חוט ישיר, בלי שקעים (כבמקור).
List<List<dynamic>> familiesImportFormatRows(Map<String, dynamic> db) =>
    fif.familiesImportFormatRows(db);

/// תומכות בפורמט-הייבוא (7 עמודות) — חוט ישיר, בלי שקעים (כבמקור).
List<List<dynamic>> supportersImportFormatRows(Map<String, dynamic> db) =>
    sif.supportersImportFormatRows(db);

/// שורות ייצוא-CSV של האירועים (9 עמודות, ממוין לפי תאריך) — החוט מחווט
/// עם termOf + שרשרת-התאריך-העברי + מילון-הסוגים (EV_META). config אופציונלי כבמקור.
List<List<dynamic>> eventsCsvRows(dynamic db, [dynamic config]) =>
    ecr.eventsCsvRows(db, config, tof.termOf, _hebFull, em.evMeta);

File: new/boxes/export-rows.mjs
/** קופסת-חיבורים · שורות-הייצוא (lib-exportRows). חוזה: export-rows.contract.md
 *  שלושת יצרני-השורות של maor/src/lib/exportRows.ts — החיווט הגלוי:
 *  משפחות/תומכות = חוטים ישירים; אירועים = חוט + שרשרת-התאריך-העברי + מילון-הסוגים.
 *  ‏toCsv/ההורדה (DOM) אינם כאן — שקע של שער-הייצוא/לוח-האם. */
import { familiesImportFormatRows as __pure_familiesImportFormatRows } from '../atoms/families-import-format-rows.mjs';
import { FAMILIES_IMPORT_FORMAT_ROWS_T as __d_familiesImportFormatRows_FAMILIES_IMPORT_FORMAT_ROWS_T } from '../atoms/families-import-format-rows-strings.mjs';
// עטיפת-כריכה (מנוע-הטיהור v3): מחרוזות-הדאטה נכרכות כאן
const familiesImportFormatRows = (...a) => __pure_familiesImportFormatRows(...a, ...Array(Math.max(0, 1 - a.length)).fill(undefined), __d_familiesImportFormatRows_FAMILIES_IMPORT_FORMAT_ROWS_T);
import { supportersImportFormatRows as __pure_supportersImportFormatRows } from '../atoms/supporters-import-format-rows.mjs';
import { SUPPORTERS_IMPORT_FORMAT_ROWS_T as __d_supportersImportFormatRows_SUPPORTERS_IMPORT_FORMAT_ROWS_T } from '../atoms/supporters-import-format-rows-strings.mjs';
// עטיפת-כריכה (מנוע-הטיהור v3): מחרוזות-הדאטה נכרכות כאן
const supportersImportFormatRows = (...a) => __pure_supportersImportFormatRows(...a, ...Array(Math.max(0, 1 - a.length)).fill(undefined), __d_supportersImportFormatRows_SUPPORTERS_IMPORT_FORMAT_ROWS_T);
import { eventsCsvRows as eventsCsvRowsWire } from '../atoms/events-csv-rows.mjs';
import { termOf } from '../atoms/term-of.mjs';
import { hebDateFull } from '../atoms/heb-date-full.mjs';
import { gem as __pure_gem } from '../atoms/gematria.mjs';
import { U, T, H } from '../atoms/gematria-data.mjs';
// עטיפת-כריכה (מנוע-הטיהור v2): הדאטה נכרכת כאן — ה-API החיצוני זהה
const gem = (...a) => __pure_gem(...a, ...Array(Math.max(0, 1 - a.length)).fill(undefined), U, T, H);
import { gemYear } from '../atoms/gem-year.mjs';
import { hebParts } from '../atoms/heb-parts.mjs';
import { EV_META } from '../atoms/ev-meta.mjs';

// ── החיווט: שרשרת-התאריך-העברי — הכרעת-הקופסה (hebrew.ts:160) ──
const hebFull = (iso) => hebDateFull(iso, gem, (y) => gemYear(y, gem), hebParts);

// ── החשיפה ──
export { familiesImportFormatRows, supportersImportFormatRows };
export const eventsCsvRows = (db, config) => eventsCsvRowsWire(db, config, termOf, hebFull, EV_META);
